### PR TITLE
fixes #10492 - validate permission name should be unique and not scoped by resource_type

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: app/models/role.rb [hunk 1]
```diff
@@ -128,7 +128,7 @@ def add_permissions(permissions, options = {})
     search = options.delete(:search)
 
     collection = Permission.where(:name => permissions).all
-    raise ArgumentError, 'some permissions were not found' if collection.size != permissions.size
+    raise ArgumentError, 'some permissions were not found' if collection.size < permissions.size
```
Comment (isratrade) [inline app/models/role.rb]: @ares, can you  review please. In RHCI, 9 permission names yield 11 results from the DB.  (RHCI is adding 2 permissions with same names as permissions defined in foreman_discovery)

Comment (domcleal) [inline app/models/role.rb]: Why are you adding permissions that already exist in another plugin?  I don't understand from the issue description (empty) how you're triggering this.

**Comment by domcleal** [comment]
Please rewrite the commit message and title, it's way too long.

**Comment by ares** [comment]
Controller level permissions do not support permissions with same name, see https://github.com/theforeman/foreman/blob/develop/app/services/foreman/access_control.rb#L35. Granular permissions should work IIRC.

Long story short, we don't support two permissions with same name in Foreman, why do you need this in the first place?.

**Comment by isratrade** [comment]
ok, I will update John Matthews from the RHCI team to change the code to have permission name be unique. However, I suggest to not have uniqueness by scope in the model.

``` ruby
  validates :name, :presence => true, :uniqueness => { :scope => :resource_type }
```

Do you agree? If so, I'll send a PR to to just have

``` ruby
validates :name, :presence => true
```

File: app/models/permission.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@ class Permission < ActiveRecord::Base
   attr_accessible :name, :resource_type
 
   validates_lengths_from_database
-  validates :name, :presence => true, :uniqueness => { :scope => :resource_type }
+  validates :name, :presence => true
```
Comment (isratrade) [inline app/models/permission.rb]: re-committed with this change to remove uniqueness by scope.

**Comment by ares** [comment]
If you remove that validation, doesn't it allow the save permissions with same name? Maybe just replace the options hash with `true`. Does the role.rb patch still make sense? I'd rather see PR adding support for permissions with same the name :-) but until we have it, removing the uniqueness scope option looks reasonable to me.

**Comment by isratrade** [comment]
@ares, you're right, PR is updated.

**Comment by ares** [comment]
so I think that the change in `app/models/role.rb` no longer makes sense?

**Comment by isratrade** [comment]
@ares, updated

**Comment by ares** [comment]
Merged as 04ad52f78f83954bec062a48a0ee3cd7d35bb9dd, thanks @isratrade

**Pull request closed without merging**